src/lib/mdns/Resolver.h

```diff
@@ -100,6 +102,65 @@ struct DiscoveredNodeData
     DiscoveredNodeData() { Reset(); }
     bool IsHost(const char * host) const { return strcmp(host, hostName) == 0; }
     bool IsValid() const { return !IsHost("") && ipAddress[0] != chip::Inet::IPAddress::Any; }
+    void Log() const
```

Comment:
This may cause sufficient logging to overflow some logs in case of large number of nodes present (e.g. 100+). The method should not use `ChipLogProgress`, but rather `ChipLogDetail` (which is the verbose logging level)

Furthermore, a new logging module like `DiscoveryVerbose` could be added, so that it can be disabled if needed. This will avoid embedded targets from all using this code by default, and building it by default.

This is used only by your new code, but added in a way that makes it enticing to use during every discovery, which may/will impact performance on some targets if used more widely, especially in presence of large number of nodes.

Suggest overall to replace the Log() method here with a `Log(const DiscoveredNodeData &)` in your own example/application, since the format seems specific to the usecase here.

Reply:
Note that there are other places in the code that can reduce repetition by just calling this function. Here is one: https://github.com/project-chip/connectedhomeip/blob/master/src/controller/python/ChipDeviceController-ScriptBinding.cpp#L336-L367

Agreed with the first couple of suggestions here: Switched to ChipLogDetail and the new DiscoveryVerbose module